src/libs/ReportSecondaryActionUtils.ts

```diff
@@ -182,51 +209,73 @@ function isMarkAsExportedAction(report: Report, policy: Policy): boolean {
-    const syncEnabled = isAutoSyncEnabled(policy);
+    const connectedIntegration = getConnectedIntegration(policy);
+    const syncEnabled = hasIntegrationAutoSync(policy, connectedIntegration);
     const isReportFinished = isReportClosedOrApproved || isReportReimbursed;
 
-    if (isAdmin && isReportFinished && hasAccountingConnection && syncEnabled) {
-        return true;
+    if (!isReportFinished || !syncEnabled) {
+        return false;
     }
 
+    const isAdmin = policy?.role === CONST.POLICY.ROLE.ADMIN;
+
     const isExporter = isPrefferedExporter(policy);
 
-    if (isExporter && isReportFinished && hasAccountingConnection && !syncEnabled) {
-        return true;
+    return isAdmin || isExporter;
+}
+
+function isHoldAction(report: Report, reportTransactions: Transaction[]): boolean {
+    const isOneExpenseReport = reportTransactions.length === 1;
+    const transaction = reportTransactions.at(0);
+
+    if (!isOneExpenseReport || !transaction) {
+        return false;
     }
 
-    return false;
+    const isTransactionOnHold = isHoldActionForTransation(report, transaction);
+    return isTransactionOnHold;
 }
 
-function isHoldAction(report: Report, reportTransactions: Transaction[]): boolean {
+function isHoldActionForTransation(report: Report, reportTransaction: Transaction): boolean {
     const isExpenseReport = isExpenseReportUtils(report);
 
     if (!isExpenseReport) {
         return false;
     }
 
-    const isReportOnHold = reportTransactions.some(isOnHoldTransactionUtils);
+    const isReportOnHold = isOnHoldTransactionUtils(reportTransaction);
 
     if (isReportOnHold) {
         return false;
     }
 
     const isOpenReport = isOpenReportUtils(report);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+
+    if (isOpenReport && isSubmitter) {
+        return true;
+    }
+
     const isProcessingReport = isProcessingReportUtils(report);
-    const isReportApproved = isReportApprovedUtils({report});
 
-    return isOpenReport || isProcessingReport || isReportApproved;
+    return isProcessingReport;
 }
 
-function isChangeWorkspaceAction(report: Report, policy: Policy, reportTransactions: Transaction[], violations: OnyxCollection<TransactionViolation[]>): boolean {
+function isChangeWorkspaceAction(report: Report, reportTransactions: Transaction[], violations: OnyxCollection<TransactionViolation[]>, policy?: Policy): boolean {
     const isExpenseReport = isExpenseReportUtils(report);
     const isReportSubmitter = isCurrentUserSubmitter(report.reportID);
-    const areWorkflowsEnabled = policy.areWorkflowsEnabled;
+    const areWorkflowsEnabled = !!(policy && policy.areWorkflowsEnabled);
     const isClosedReport = isClosedReportUtils(report);
 
+    const policies = getAllPolicies();
+    const currentUserEmail = getCurrentUserEmail();
+    const policiesEligibleForChange = policies.filter((newPolicy) => isWorkspaceEligibleForReportChange(newPolicy, report, policy, currentUserEmail));
```

Comment:
@luacmartins I had to add this condition to avoid this situation: 
<img width="903" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/5750dfe1-3a12-499f-b777-890a88dcc1b6" />